server/re/src/icatGeneralMS.cpp

```diff
@@ -138,11 +138,14 @@ msiQuota( ruleExecInfo_t *rei ) {
  * \sa none
 **/
 int  msiSetResource( msParam_t* xrescName, ruleExecInfo_t *rei ) {
-    char *rescName;
+    char* rescName{static_cast<char *>(xrescName->inOutStruct)};
 
-    rescName = ( char * ) xrescName->inOutStruct;
+    if (rei->doi == nullptr) {
+        rodsLog(LOG_ERROR, "Cannot set resource, DOI not found.");
+        return ACTION_FAILED_ERR;
```

Comment:
Leave yourself a comment about updating this error code so you don't forget.

Reply:
Done.